test/IdTableTest.cpp

```diff
@@ -264,11 +266,11 @@ auto clone(const auto& table, auto... args) {
   }
 }
 
-TEST(IdTable, push_back_and_assign) {
-  // A lambda that is used as the `testCase` argument to the
-  // `runTestForDifferentTypes` function (see above for details).
-  auto runTestForIdTable = []<typename Table>(auto make,
-                                              auto... additionalArgs) {
+// A callable that is used as the `testCase` argument to the
+// `runTestForDifferentTypes` function (see above for details).
+struct CheckPushBackAndAssign {
```

Comment:
I think here we can also rewrite the infrastructure to use the `Type_identity` trick.

Reply:
Addressed